File: level2/소수_찾기.cpp
#include <string>
#include <vector>
#include <cstdlib>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> primeNumbers;

void findPrimeNumber(string numbers, int number){
    if(number != 0 && number != 1){
        bool Prime = true;
        for(int i=2; i<=number/2; i++){
          if(number%i == 0){
              Prime = false;
              break;
          }
        }
        if(Prime == true)
            primeNumbers.push_back(number);
    }

    for(int i=0; i<numbers.size();i++){
        string temp = numbers;
        temp.erase (temp.begin()+i);
        findPrimeNumber(temp, number*10 + (int)numbers[i]  - 48);
    }
}

int solution(string numbers) {

    findPrimeNumber(numbers, 0);
    sort(primeNumbers.begin(), primeNumbers.end());
    primeNumbers.erase(unique(primeNumbers.begin(), primeNumbers.end()), primeNumbers.end());

    return primeNumbers.size();
}

File: JavaScript/level2/소수_찾기.js
let cache = [];

function PrimeNumber(val){
    if(val === 0 || val === 1)
        return false;

    for(let i=2; i<=val/2; i++)
        if(val%i === 0)
            return false;
    
    return true;
}

function FindPrimrNumber(str, val){
    if(PrimeNumber(val))
        cache.push(val);
    
    for(let i=0; i<str.length; i++)
        FindPrimrNumber(str.slice(0, i) + str.slice(i+1), val*10 + Number(str[i]));
}

function solution(numbers) {
    FindPrimrNumber(numbers, 0);

    return [...new Set(cache)].length;
}